crates/examples/combined/all.rs

```diff
@@ -140,12 +141,19 @@ async fn main() {
             8000 + (u16::try_from(i).expect("failed to create advertise address")),
         );
 
+        let builder_address = Url::parse(&format!(
+            "http://localhost:{}",
+            9000 + (u16::try_from(i).expect("failed to create builder address"))
+        ))
+        .unwrap();
+
```

Comment:
Looks like instead of `localhost`, we need to extract the machines local ip, so that when nodes are running on different machines they could know each other's address.